src/Tools/ExternalAccess/Razor/Features/RazorAnalyzerAssemblyRedirector.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Composition;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Workspaces.AnalyzerRedirecting;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.Razor;
+
+[Export(typeof(IAnalyzerAssemblyRedirector)), Shared]
+[method: ImportingConstructor]
+[method: Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+internal sealed class RazorAnalyzerAssemblyRedirector([Import(AllowDefault = true)] RazorAnalyzerAssemblyRedirector.IRazorAnalyzerAssemblyRedirector? razorRedirector = null) : IAnalyzerAssemblyRedirector
```

Comment:
Nit: Line is getting a little long here and wrapping, consider breaking it up a bit.